lib/cli_test.go

```diff
@@ -0,0 +1,60 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package lib
+
+import (
+	"flag"
+	"testing"
+
+	errwrap "github.com/pkg/errors"
+
+	"github.com/urfave/cli"
+)
+
+// TestValidatePass tests if lint passes on a valid file
+func TestValidatePass(t *testing.T) {
+	set := flag.NewFlagSet("test", 0)
+	set.String("lang", "../examples/lang/load0.mcl", "doc")
+	ctx := cli.NewContext(nil, set, nil)
+	if err := validate(ctx); err != nil {
+		t.Fatal(errwrap.Wrapf(err, "example load0.mcl should pass validation"))
+	}
+}
+
+// TestValidateError tests if validation fails on files with invalid syntax
+func TestValidateError(t *testing.T) {
+	files := []struct {
+		path string
+	}{
+		{"../examples/lang/bad/syntaxerror.mcl"},
```

Comment:
no ../ paths please and no references to examples/ in our code.
we need to think of a better way to get the code in. i'd actually prefer to just inline it for now. see:

https://github.com/purpleidea/mgmt/blob/master/lang/lexparse_test.go#L93